Add `toSinglyLinkedRing` and `toDoublyLinkedRing` to `std/lists`

ericvw opened the pull request:
Allow for conversion from `openArray`s, similar to `toSinglyLinkedList` and `toDoublyLinkedList`.

Varriount commented on lib/pure/collections/lists.nim:
```diff
@@ -983,6 +983,16 @@ func toSinglyLinkedList*[T](elems: openArray[T]): SinglyLinkedList[T] {.since: (
+    let a = [1, 2, 3, 4, 5].toSinglyLinkedRing
+    assert a.toSeq == [1, 2, 3, 4, 5]
+
+  for elem in elems.items:
```
Shouldn't this call `initSinglyLinkedRing`?

ericvw replied:
I don't believe so; according to: https://github.com/nim-lang/Nim/blob/cd84cd45ea5bc1de67dd28dab906ad6c16e7b092/lib/pure/collections/lists.nim#L113-L114
Also, the implementation is `discard`.

If there is some subtle reason or stylistic preference, I'm happy to update accordingly.

Araq replied:
Use `initSinglyLinkedRing` to have the code future-proof.

ericvw replied:
Ack. I'll update to use `initSinglyLinkedRing`.

Araq commented on lib/pure/collections/lists.nim:
```diff
@@ -983,6 +983,16 @@ func toSinglyLinkedList*[T](elems: openArray[T]): SinglyLinkedList[T] {.since: (
   for elem in elems.items:
     result.add(elem)
 
+func toSinglyLinkedRing*[T](elems: openArray[T]): SinglyLinkedRing[T] {.since: (2, 0, 0).} =
```
Why would this be `since: (2, 0, 0)`?

ericvw replied:
I was mimicking the `toSinglyLinkedList` declaration.

I guess I don't fully understand when `since` should be used. Should I remove the `since`, or should the version be something else?

ericvw committed:
Add `toSinglyLinkedRing` and `toDoublyLinkedRing` to `std/lists`

Allow for conversion from `openArray`s, similar to `toSinglyLinkedList`
and `toDoublyLinkedList`.

Araq commented on lib/pure/collections/lists.nim:
```diff
@@ -983,6 +983,17 @@ func toSinglyLinkedList*[T](elems: openArray[T]): SinglyLinkedList[T] {.since: (
   for elem in elems.items:
     result.add(elem)
 
+func toSinglyLinkedRing*[T](elems: openArray[T]): SinglyLinkedRing[T] {.since: (2, 0, 0).} =
```
```suggestion
func toSinglyLinkedRing*[T](elems: openArray[T]): SinglyLinkedRing[T] =
```

Araq commented on lib/pure/collections/lists.nim:
```diff
@@ -993,3 +1004,14 @@ func toDoublyLinkedList*[T](elems: openArray[T]): DoublyLinkedList[T] {.since: (
   for elem in elems.items:
     result.add(elem)
+
+func toDoublyLinkedRing*[T](elems: openArray[T]): DoublyLinkedRing[T] {.since: (2, 0, 0).} =
```
```suggestion
func toDoublyLinkedRing*[T](elems: openArray[T]): DoublyLinkedRing[T] =
```
